mlflow/pyfunc/model.py

```diff
@@ -593,32 +574,7 @@ def _convert_input(self, model_input):
         import pandas as pd
 
         hints = self.python_model._get_type_hints()
-        if _is_list_str(hints.input):
-            if isinstance(model_input, pd.DataFrame):
-                first_string_column = _get_first_string_column(model_input)
-                if first_string_column is None:
-                    raise MlflowException.invalid_parameter_value(
-                        "Expected model input to contain at least one string column"
-                    )
-                return model_input[first_string_column].tolist()
-            elif isinstance(model_input, list):
-                if all(isinstance(x, dict) for x in model_input):
-                    return [next(iter(d.values())) for d in model_input]
```

Comment:
Not sure why we supported passing list[dict] to list[str] type hint before, but it's not correct and we won't support this anymore.